File: java/src/main/java/edu/cmu/cs/cs214/lab02/shapes/Square.java
package edu.cmu.cs.cs214.lab02.shapes;

public final class Square implements Shape {
    private double sideLen;

    public Square(double sideLen) {
        this.sideLen = sideLen;
    }
    public Square(Square o1) {
        this.sideLen = o1.sideLen;
    }
    @Override
    public double getArea() {
        return sideLen * sideLen;
    }
}

File: typescript/src/shapes/square.ts
import Shape from "./shape";

export default class Square implements Shape {
    #sideLen:number;
    constructor(s:number) {
        this.#sideLen = s;
    }
    computeArea = () :number => {
        return this.#sideLen * this.#sideLen;
    }
}